xbmc/video/VideoInfoScanner.cpp

```diff
@@ -446,6 +452,120 @@ namespace KODI::VIDEO
+                              movieDetails.GetUpdateSetOverview()); // Update set
+            CGUIListItem::ArtMap movieSetArt;
+            auto urls = movieDetails.m_strPictureURL.GetUrls();
+            for (auto& url : urls)
+              if (StringUtils::StartsWith(url.m_aspect, "set."))
+                movieSetArt.insert({url.m_aspect.substr(4), url.m_url});
+            m_database.SetArtForItem(dbId, MediaTypeVideoCollection, movieSetArt);
+          }
+        }
+      }
+    }
+  }
+
+  bool CVideoInfoScanner::UpdateSetInTag(CVideoInfoTag& tag)
+  {
+    // Uses the set information in m_set from a tag of a movie
+    // If there is a set, see if the details need to be updated from the
+    // Movie Set Information Folder
+    bool setUpdated{false};
+    if (tag.m_set.HasTitle())
+    {
+      const std::string movieSetInfoPath = GetMovieSetInfoFolder(tag.m_set.GetTitle());
+      if (!movieSetInfoPath.empty())
+      {
+        // look for Set.NFO
+        if (const std::unique_ptr setLoader{
+                CSetInfoTagLoaderFactory::CreateLoader(tag.m_set.GetTitle())};
+            setLoader)
+        {
+          CSetInfoTag setTag;
+          INFO_TYPE result{setLoader->Load(setTag, false)};
+          if (result && !setTag.IsEmpty())
+          {
+            tag.m_set.SetOriginalTitle(tag.m_set.GetTitle());
+            if (!setTag.GetTitle().empty())
+              tag.m_set.SetTitle(setTag.GetTitle());
+            if (!setTag.GetOverview().empty())
+            {
+              tag.m_set.SetOverview(setTag.GetOverview());
+              tag.SetUpdateSetOverview(true);
+            }
+            if (!setTag.GetPoster().empty())
+              tag.m_set.SetPoster(setTag.GetPoster());
+            setUpdated = true;
+          }
+        }
+
+        // Now look for art
+        // If poster specified in set.nfo use that first
+        CGUIListItem::ArtMap movieSetArt;
+        if (tag.m_set.HasPoster())
+        {
+          movieSetArt.insert({"poster", tag.m_set.GetPoster()});
+        }
+        else
+        {
+          const std::vector<std::string> movieSetArtTypes =
+              CVideoThumbLoader::GetArtTypes(MediaTypeVideoCollection);
+          AddLocalItemArtwork(movieSetArt, movieSetArtTypes, movieSetInfoPath, true, false);
+        }
+        if (!movieSetArt.empty())
+        {
+          tag.m_strPictureURL.Parse();
+          auto urls = tag.m_strPictureURL.GetUrls();
+
+          // Clear existing art (if any) from cache and VideoInfoTag
+          CTextureDatabase textureDb;
+          if (textureDb.Open())
+          {
+            for (const auto& url : urls)
+              if (StringUtils::StartsWith(url.m_aspect, "set."))
+                textureDb.InvalidateCachedTexture(url.m_url);
+            textureDb.Close();
+          }
+
+          // Remove existing set art from VideoInfoTag
+          std::erase_if(urls, [](const auto& url)
```

Comment:
This method shouldn't be using `m_strPictureURL` at all, I figure. That is a list of all available artwork returned from the scraper; it contains not just multiple artwork types, like "poster", "clearlogo", "set.poster", and "set.fanart", but also multiple options for each type. These options can be selected in the GUI. We don't want to remove options from that list.

You could create a separate function to update info like title and overview, and another function to return the ArtMap for the set.